lib/rubocop/cop/sorbet/multiple_t_enum_values.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Sorbet
+      # Disallow creating a `T::Enum` with less than two values.
+      #
+      # @example
+      #
+      #  # bad
+      #  class ErrorMessages < T::Enum
+      #    enums do
+      #      ServerError = new("There was a server error.")
+      #    end
+      #  end
+      #
+      #  # good
+      #  class ErrorMessages
+      #    SERVER_ERROR = "There was a server error."
+      #  end
+
+      class MultipleTEnumValues < Base
+        def initialize(*)
+          @scopes = []
+          super
+        end
+
+        MSG = "`T::Enum` should have at least two values."
+
+        # @!method t_enum?(node)
+        def_node_matcher :t_enum?, <<~PATTERN
+          (class (const...) (const (const nil? :T) :Enum) ...)
+        PATTERN
+
+        # @!method enums_block?(node)
+        def_node_matcher :enums_block?, <<~PATTERN
+          (block (send nil? :enums) ...)
+        PATTERN
+
+        def on_class(node)
+          @scopes.push(node) if t_enum?(node)
+        end
+
+        def after_class(node)
+          @scopes.pop if t_enum?(node)
+        end
+
+        def on_block(node) # rubocop:disable InternalAffairs/NumblockHandler
+          return if @scopes.empty?
+          return unless enums_block?(node)
+
+          scope = @scopes.last
+
+          if node.body.nil?
+            add_offense(scope)
+            return
+          end
+
+          begin_node = node.children.find(&:begin_type?)
+
+          num_casgn_nodes = if begin_node
+            begin_node.children.count(&:casgn_type?)
+          else
+            node.children.count(&:casgn_type?)
+          end
```

Comment:
Not sure if this would be easier to read?

```suggestion
          num_casgn_nodes = (begin_node || node).children.count(&:casgn_type?)
```

Or something like this?

```suggestion
          node = begin_node if begin_node
          node.children.count(&:casgn_type?)
```

In any case, let's remove the `children.count(&:casgn_type?)` duplication.